clash-lib/src/Clash/Core/TermLiteral/TH.hs

```diff
@@ -33,6 +42,7 @@ deriveTermToData typName = do
   pure (deriveTermToData1 (map toConstr' constrs))
  where
   toConstr' (NormalC cName fields) = (cName, length fields)
+  toConstr' (RecC cName fields) = (cName, length fields)
```

Comment:
Do we need a Changelog entry saying we can now process records and type-level naturals?

Reply:
Yes

Reply:
Done! But how does this handle type-level naturals differently than before?